web/styles/rendered_markdown.css

```diff
@@ -23,15 +23,39 @@
         margin-left: 0;
     }
 
-    /* Ensure ordered lists are nicely centered in 1-line messages */
     & ol {
-        margin: 2px 0 5px 20px;
+        margin: 2px 0 5px;
+
+        /* Ensure ordered lists are nicely centered in 1-line messages for up to 2
+           digit markers, else the markers take up as much space as needed without
+           being cut off at the beginning */
+        > li {
+            display: grid;
+            grid-template-columns: minmax(20px, max-content) 1fr;
+
+            &::before {
+                content: counter(list-item) ". ";
+                text-align: right;
+                counter-increment: list-item;
```

Comment:
Small style nit, but put the increment line net to the content line, and keep the text and margin properties together.